scapy/layers/dns.py

```diff
@@ -43,6 +43,8 @@ def h2i(self, pkt, x):
     def i2m(self, pkt, x):
         if x == b".":
             return b"\x00"
+        elif isinstance(x, str):
```

Comment:
I'm not sure when/if this can happen, as "internal" value should always be bytes.